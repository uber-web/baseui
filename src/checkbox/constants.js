// @flow
/* eslint-disable import/prefer-default-export */

export const STATE_TYPE = {
  change: 'CHANGE',
  focus: 'FOCUS',
  blur: 'BLUR',
  hover: 'HOVER',
  unhover: 'UNHOVER',
};

/* eslint-enable import/prefer-default-export */
