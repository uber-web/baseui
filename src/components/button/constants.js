/* eslint-disable flowtype/require-valid-file-annotation */

export const BUTTON_TYPE = {
  square: 'square',
  round: 'round',
};
