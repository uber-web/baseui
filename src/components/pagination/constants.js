/* eslint-disable flowtype/require-valid-file-annotation */
/* eslint-disable import/prefer-default-export */

export const STATE_CHANGE_TYPE = {
  prevPage: 'prevPage',
  nextPage: 'nextPage',
};
