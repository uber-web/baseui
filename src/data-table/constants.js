/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const COLUMNS = Object.freeze({
  ANCHOR: 'ANCHOR',
  CATEGORICAL: 'CATEGORICAL',
  NUMERICAL: 'NUMERICAL',
  BOOLEAN: 'BOOLEAN',
  STRING: 'STRING',
  CUSTOM: 'CUSTOM',
});

export const NUMERICAL_FORMATS = Object.freeze({
  DEFAULT: 'DEFAULT',
  ACCOUNTING: 'ACCOUNTING',
  PERCENTAGE: 'PERCENTAGE',
});

export const NUMERICAL_OPERATIONS = Object.freeze({
  EQ: 'EQ',
  GT: 'GT',
  GTE: 'GTE',
  LT: 'LT',
  LTE: 'LTE',
});

export const SORT_DIRECTIONS = Object.freeze({
  ASC: 'ASC',
  DESC: 'DESC',
});
