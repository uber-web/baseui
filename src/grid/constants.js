/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const BEHAVIOR = Object.freeze({
  fluid: 'fluid',
  fixed: 'fixed',
});

export const ALIGNMENT = Object.freeze({
  start: 'flex-start',
  center: 'center',
  end: 'flex-end',
});

export const DEFAULT_GRID_COLUMNS = [4, 8, 12];
export const DEFAULT_GRID_GUTTERS = [16, 36, 36];
export const DEFAULT_GRID_MARGINS = [16, 36, 64];
export const DEFAULT_GRID_MAX_WIDTH = 1280;
