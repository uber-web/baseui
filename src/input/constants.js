/* eslint-disable import/prefer-default-export */
export const STATE_TYPE = {
  change: 'change',
  focus: 'focus',
  blur: 'blur',
};
/* eslint-enable import/prefer-default-export */
