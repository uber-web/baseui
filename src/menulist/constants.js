// @flow
/* eslint-disable import/prefer-default-export */

export const STATE_CHANGE_TYPES = {
  keyPressArrowUp: 'keyPressArrowUp',
  keyPressArrowDown: 'keyPressArrowDown',
};

// Dict of props to prepend $ to pass thru to styletron
export const STYLETRON_PROP_MAPPER = {
  ref: true,
  isHighlighted: true,
};

export const KEY_STRINGS = {
  ArrowUp: 'ArrowUp',
  ArrowDown: 'ArrowDown',
  Enter: 'Enter',
};
