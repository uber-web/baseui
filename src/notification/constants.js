/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const KIND = Object.freeze({
  primary: 'primary',
  success: 'success',
  warning: 'warning',
  error: 'error',
});
