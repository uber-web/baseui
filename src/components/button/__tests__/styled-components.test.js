/* eslint-disable flowtype/require-valid-file-annotation, react/prop-types */
import React from 'react';
import {shallow} from 'enzyme';
import {
  BaseButton,
  PrimaryButton,
  SecondaryButton,
  TertiaryButton,
  MinimalButton,
  ButtonLabel,
  StartEnhancer,
  EndEnhancer,
} from '../styled-components';
import {BUTTON_KIND, BUTTON_SIZE} from '../constants';

function makeTest({
  title,
  component: Component,
  props = {},
  children = null,
  snapshotName = 'correct styles',
}) {
  test(title, () => {
    const shallowed = shallow(<Component {...props}>{children}</Component>);
    expect(shallowed.instance().getStyles()).toMatchSnapshot(snapshotName);
  });
}

describe('Button Styled Components', () => {
  makeTest({
    title: 'BaseButton - basic render',
    component: BaseButton,
  });

  makeTest({
    title: 'BaseButton - round',
    component: BaseButton,
    props: {
      $kind: BUTTON_KIND.round,
    },
  });

  makeTest({
    title: 'BaseButton - round (compact)',
    component: BaseButton,
    props: {
      $type: BUTTON_SIZE.compact,
      $kind: BUTTON_KIND.round,
    },
  });

  makeTest({
    title: 'PrimaryButton - basic render',
    component: PrimaryButton,
  });

  makeTest({
    title: 'SecondaryButton - basic render',
    component: SecondaryButton,
  });

  makeTest({
    title: 'TertiaryButton - basic render',
    component: TertiaryButton,
  });

  makeTest({
    title: 'MinimalButton - basic render',
    component: MinimalButton,
  });

  makeTest({
    title: 'ButtonLabel - basic render',
    component: ButtonLabel,
  });

  makeTest({
    title: 'StartEnhancer - basic render',
    component: StartEnhancer,
  });

  makeTest({
    title: 'EndEnhancer - basic render',
    component: EndEnhancer,
  });
});
