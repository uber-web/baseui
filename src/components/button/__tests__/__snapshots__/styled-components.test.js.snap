// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Styled Components BaseButton - basic render: correct styles 1`] = `
Object {
  ":disabled": Object {
    "backgroundColor": "$theme.colors.mono300",
    "color": "$theme.colors.mono600",
    "cursor": "not-allowed",
  },
  "WebkitAppearance": "none",
  "alignItems": "center",
  "border": "none",
  "borderRadius": "$theme.borders.radius200",
  "cursor": "pointer",
  "display": "flex",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "justifyContent": "space-between",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "outline": "none",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale300",
}
`;

exports[`Button Styled Components BaseButton - round: correct styles 1`] = `
Object {
  ":disabled": Object {
    "backgroundColor": "$theme.colors.mono300",
    "color": "$theme.colors.mono600",
    "cursor": "not-allowed",
  },
  "WebkitAppearance": "none",
  "alignItems": "center",
  "border": "none",
  "borderRadius": "50%",
  "cursor": "pointer",
  "display": "flex",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "justifyContent": "space-between",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "outline": "none",
  "padding": "$theme.sizing.scale500",
}
`;

exports[`Button Styled Components ButtonLabel - basic render: correct styles 1`] = `Object {}`;

exports[`Button Styled Components EndEnhancer - basic render: correct styles 1`] = `
Object {
  "marginLeft": "$theme.sizing.scale500",
}
`;

exports[`Button Styled Components MinimalButton - basic render: correct styles 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "$theme.colors.mono400",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.mono200",
  },
  "backgroundColor": "transparent",
  "color": "$theme.colors.primary",
}
`;

exports[`Button Styled Components PrimaryButton - basic render: correct styles 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "$theme.colors.primary600",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.primary500",
  },
  "backgroundColor": "$theme.colors.primary",
  "color": "$theme.colors.white",
}
`;

exports[`Button Styled Components SecondaryButton - basic render: correct styles 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "$theme.colors.primary200",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.primary100",
  },
  "backgroundColor": "$theme.colors.primary50",
  "color": "$theme.colors.primary",
}
`;

exports[`Button Styled Components StartEnhancer - basic render: correct styles 1`] = `
Object {
  "marginRight": "$theme.sizing.scale500",
}
`;

exports[`Button Styled Components TertiaryButton - basic render: correct styles 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "$theme.colors.mono500",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.mono400",
  },
  "backgroundColor": "$theme.colors.mono200",
  "color": "$theme.colors.primary",
}
`;
